builder/azure/arm/resource_resolver_test.go

```diff
@@ -0,0 +1,75 @@
+package arm
+
+import (
+	"testing"
+)
+
+func TestResourceResolverIgnoresEmptyVirtualNetworkName(t *testing.T) {
+	c, _, _ := newConfig(getArmBuilderConfiguration(), getPackerConfiguration())
+	if c.VirtualNetworkName != "" {
+		t.Fatalf("Expected VirtualNetworkName to be empty by default")
+	}
+
+	sut := newTestResourceResolver()
+	sut.findVirtualNetworkResourceGroup = nil // assert that this is not even called
+	sut.Resolve(c)
+
+	if c.VirtualNetworkName != "" {
+		t.Fatalf("Expected VirtualNetworkName to be empty")
+	}
+	if c.VirtualNetworkResourceGroupName != "" {
+		t.Fatalf("Expected VirtualNetworkResourceGroupName to be empty")
+	}
+}
+
+// If the user fully specified the virtual network name and resource group then
+// there is no need to do a lookup.
+func TestResourceResolverIgnoresSetVirtualNetwork(t *testing.T) {
+	c, _, _ := newConfig(getArmBuilderConfiguration(), getPackerConfiguration())
+	c.VirtualNetworkName = "--virtual-network-name--"
+	c.VirtualNetworkResourceGroupName = "--virtual-network-resource-group-name--"
+	c.VirtualNetworkSubnetName = "--virtual-network-subnet-name--"
+
+	sut := newTestResourceResolver()
+	sut.findVirtualNetworkResourceGroup = nil // assert that this is not even called
+	sut.Resolve(c)
```

Comment:
Does it make sense to also set `sut.findVirtualNetworkSubnet=nil` to check that it doesn't get called since the subnet name is user-supplied?